### PR TITLE
AO3-4332 Fix mobile filters on unassigned fandoms page

### PR DESCRIPTION
**Pull request by sarken**
https://otwarchive.atlassian.net/browse/AO3-4332

This makes it so the filters on the unassigned fandoms page can be shown/hidden like the filters on other pages. It also fixes a display issue where, when JavaScript was disabled, the filters would not correctly display at 100% width under the list of fandoms.

If PR #2181 is merged first, the CSS change in this will need to be updated. Likewise, if this is merged first, PR #2181 will need its CSS updated.

### CODE REVIEW

File: app/views/fandoms/unassigned.html.erb [hunk 1]
```diff
@@ -20,10 +29,13 @@
...
 
 <!--subnav-->
-<%= form_tag unassigned_fandoms_path, :class => 'filters', :method => :get do %>
+<%= form_tag unassigned_fandoms_path,
+    method: :get,
+    class: 'narrow-hidden filters',
+    id: 'fandom-filters' do %>
   <%= field_set_tag ts('Filter results:') do %>
     <h3 class="landmark heading"><%= ts('Filters') %></h3>
-    <dl class="filters" role="menu">
```
Comment (sarken) [inline app/views/fandoms/unassigned.html.erb]: role="menu" isn't supposed to be used this way, so out it goes.

File: app/views/fandoms/unassigned.html.erb [hunk 2]
```diff
@@ -3,9 +3,18 @@
 <!--/descriptions-->
 
 <!--subnav-->
-<%= will_paginate(@fandoms) %>
+<% # Filters button for narrow screens jumps to filters when JavaScript is disabled and opens filters when JavaScript is enabled %>
+<ul class="navigation actions" role="navigation">
+  <li class="narrow-shown hidden">
+    <a href="#fandom-filters" id="go_to_filters">
+      <%= ts("Filters") %>
```
Comment (cesy) [inline app/views/fandoms/unassigned.html.erb]: Should we be using the new translation system here, t rather than ts?

Comment (sarken) [inline app/views/fandoms/unassigned.html.erb]: I don't think so. We haven't discussed how exactly we're going to do it (e.g. naming conventions for keys), and any time we try to use it, the pull request ends up being delayed for an eternity. So we've been sticking with the current system.

**Comment by sarken** [comment]
Got rid of the merge conflicts.

**Comment by cesy** [comment]
Looks good to me.

**Pull request merged**